### PR TITLE
Resolve "Active/Inactive filter buttons of output dock widget are hard to distinguish in dark mode"

### LINKED ISSUE #606: Active/Inactive filter buttons of output dock widget are hard to distinguish in dark mode
In GitLab by @schn_bo on Oct 13, 2023, 06:23

## What is the purpose of the change?
- In the black mode of the GTlab the settings of the output are not well distinguishable. It would be great if activated output setting would be stronger emphasis. Moreover, I can not really see a difference between the "Test-Output"-button and the "Show-Hide-Debug-Output"-button. (maybe some spacer between the output visibility and the remaining button could lead to an better useabiltiy.

grafik

### PR DESCRIPTION
**Pull request by mariusalexander**
## Description

Closes #606.

Applied a highlighted effect to the filter buttons if they are enabled

grafik

## How Has This Been Tested?

Manually

## Checklist:

- [ ] A test for the new functionality was added (if applicable).
- [ ] All tests run without failure.
- [ ] The changelog has been extended, if this MR contains important changes from the users's point of view.
- [ ] The new code complies with the GTlab's style guide.
- [ ] New interface methods / functions are exported via `EXPORT`. Non-interface functions are NOT exported.
- [ ] The number of code quality warnings is not increasing.

### CODE REVIEW

File: src/app/dock_widgets/output/gt_outputdock.cpp [hunk 1]
```diff
@@ -184,16 +184,31 @@ GtOutputDock::GtOutputDock()
...
         auto* button = setupButton(icon, tooltip);
         button->setCheckable(true);
 
+        // checked button do not use On/Off Icons, thus we have to update the
+        // icon ourselfes
+        auto const updateIconColor = [b = QPointer<QPushButton>(button)](){
+            assert (b);
+            return b->isChecked() ? highlight() : disabled();
+        };
+        button->setIcon(colorize(icon, gt::gui::SvgColorData{ updateIconColor }));
```
Comment (mariusalexander) [inline src/app/dock_widgets/output/gt_outputdock.cpp:206]: This is where the colorization of the button is happening

**Comment by mariusalexander** [comment]
> @mariusalexander We decided, that we only want one reviewer for PRs to avoid the diffusion of responsibilities. Who should review it?

Thanks for the info, but I think I have not yet requested a review from anybody yet for this issue. Do you mean PR ttps://github.com/dlr-gtlab/gtlab-core/pull/1166?

**Comment by rainman110** [comment]
> > @mariusalexander We decided, that we only want one reviewer for PRs to avoid the diffusion of responsibilities. Who should review it?
> 
> Thanks for the info, but I think I have not yet requested a review from anybody yet for this issue. Do you mean PR #1166?

Ah, my fault. I was just misinterpreting the Reviewers section 😆 
image

**Comment by mariusalexander** [comment]
@rainman110 But then I will request an review from you. I am still experimenting with what colors do work best, though. For bright mode I like the normal text colors: (on windows! I think on linux in the "Fusion" style there is no blueish highlight within the button)

grafik

That is how the text colors would look like in dark mode:

grafik

When I use the "highlighted" color in bright mode:

grafik

and in dark mode:

grafik



I have also added an spacer between the buttons on the left and the filter buttons. Is it okay that way?

**Comment by rainman110** [comment]
Not a real review: 

I just checked it out in windows. I don't really like the blue color, as it attracts attention. I would rather prefer an active button to be dark (in bright mode) and greyed out  if inactive.

In dark mode, an active button should be white and an inactive button some darker grey tone.

**Comment by mariusalexander** [comment]
I further lighten/darkened the disabled color to make the difference a bit bigger

grafik

grafik

It is ready to merge

**Comment by mariusalexander** [comment]
> _Obviously this is a matter of taste!_
> 
> _And, it's better than before!!!_

I agree, it is 100% personal preference. Glad you like it

**Review by rainman110 — APPROVED** [review]
I like it. Great work 👍

**Pull request merged**